backend/src/main.js

```diff
@@ -1,114 +1,178 @@
+            res.body = "Login failed";
+            return res.status(401).json({ status: 'error', code: 'unauthorized' });
+        } else {
+            const payload = {
+                _id: user.ID,
+                displayName: user.username,
+                email: user.email
+            };
+            const token = jwt.sign(payload, "mysecretkey");
+            return res.json({user: user.username, access_token: token});
+        }
+    })(req, res, next);
+});
+
+/*app.post('/api/token', async function (req, res, next) {
+    if (req.body.user && req.body.password) {
+        var username = req.body.user;
+        var password = req.body.password;
+        var usertest = user.find(function(u) {
+            return u.email === email && u.password === password;
+        });
+        if (usertest) {
+            var payload = {
+                id: usertest._id,
+                displayName: usertest.username,
+                email: usertest.email
+            };
+            var token = jwt.sign(payload, "mysecretkey");
+            res.json({
+                token: token
+            });
+        } else {
+            res.sendStatus(401);
+        }
+    } else {
+        res.sendStatus(401);
+    }
+});*/
+
+app.delete('/api/token',  function (req, res, next) {
+    return res.location(`/home`).send();
+});
+
+app.post('/api/users', async function (req, res, next) {
+    try {
+        await user.create({
+            username: req.body.username,
+            password: req.body.password,
+            email: req.body.email});
+        log.info("new user entry created");
+        res.status = 201;
+        res.location(`/api/login`);
+        res.send();
+    } catch(err) {
+        if(err.username == 'ValidationError') {
+            res.statusCode = 400;
+            res.send({ error: 'Validation error' });
+        } else {
             res.statusCode = 500;
-            log.error('Internal error(%d): %s',res.statusCode,err.message);
-            return res.send({ error: 'Server error' });
+            res.send({ error: 'Server error' });
         }
-        return res.send(passentry)
-    });
+        log.error('Internal error(%d): %s',res.statusCode,err.message);
+    }
 });
 
 app.post('/api/entries', function (req, res, next) {
-    let entryid = guid();
-    console.log(req.body);
-    passEntry.create({
-        id: entryid,
-        title: req.body.title,
-        user: req.body.user,
-        password: req.body.password
-    }, function (err) {
-        if (!err) {
-            log.info("new password entry created");
-            res.status = 201;
-            res.location(`/api/entries/${entryid}`);
+    return passport.authenticate('jwt', {session: 'false'}, async function (err, user, info){
+        try {
+            let newentryID = await passEntry.create({
+            userID: user.ID,
+            title: req.body.title,
+            user: req.body.user,
+            password: req.body.password}).get('ID');
```

Comment:
It's a habit from cache db, i'll try to name fields in db like sNote or oUser ("s" means string, "o" means object).